nx/source/services/fs.c

```diff
@@ -101,6 +101,44 @@ Result fsOpenBisStorage(FsStorage* out, u32 PartitionId) {
     return rc;
 }
 
+Result fsOpenBisFileSystem(FsFileSystem* out, u32 PartitionId, const char* string) {
+    IpcCommand c;
+    ipcInitialize(&c);
+    ipcAddSendStatic(&c, string, strlen(string) + 1, 0);
```

Comment:
strncpy(tmpstr, string, sizeof(tmpstr)-1);

Then ipcAddSendStatic with tmpstr and sizeof(tmpstr).